diff --git a/CMakeLists.txt b/CMakeLists.txt
index 97fc42a..d8e5f50 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -44,7 +44,7 @@ endif ()
 ##-----------------------------------------------------------------------------
 ##  CMake path
 ##-----------------------------------------------------------------------------
-set (CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake" ${CMAKE_MODULE_PATH})
+set (CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake" ${CMAKE_MODULE_PATH})
 
 ##-----------------------------------------------------------------------------
 ##  Prevent in-source build
diff --git a/libs/demons_itk_insight/LOGDomainDemons/itkLogDomainDeformableRegistrationFilter.hxx b/libs/demons_itk_insight/LOGDomainDemons/itkLogDomainDeformableRegistrationFilter.hxx
index ee56c27..df92ab4 100644
--- a/libs/demons_itk_insight/LOGDomainDemons/itkLogDomainDeformableRegistrationFilter.hxx
+++ b/libs/demons_itk_insight/LOGDomainDemons/itkLogDomainDeformableRegistrationFilter.hxx
@@ -3,7 +3,8 @@
 
 #include "itkLogDomainDeformableRegistrationFilter.h"
 
-#include "itkExceptionObject.h"
+// #include "itkExceptionObject.h"
+#include "itkMacro.h"
 #include "itkImageRegionIterator.h"
 #include "itkImageLinearIteratorWithIndex.h"
 #include "itkDataObject.h"
diff --git a/libs/demons_itk_insight/itkESMDemonsRegistrationWithMaskFunction.txx b/libs/demons_itk_insight/itkESMDemonsRegistrationWithMaskFunction.txx
index 321e826..8f83d00 100644
--- a/libs/demons_itk_insight/itkESMDemonsRegistrationWithMaskFunction.txx
+++ b/libs/demons_itk_insight/itkESMDemonsRegistrationWithMaskFunction.txx
@@ -19,7 +19,8 @@
 #define __itkESMDemonsRegistrationWithMaskFunction_txx
 
 #include "itkESMDemonsRegistrationWithMaskFunction.h"
-#include "itkExceptionObject.h"
+// #include "itkExceptionObject.h"
+#include "itkMacro.h"
 #include "vnl/vnl_math.h"
 #if ITK_VERSION_MAJOR >= 5
 #include "vcl_legacy_aliases.h"
diff --git a/libs/demons_itk_insight/itkPDEDeformableRegistrationWithMaskFilter.hxx b/libs/demons_itk_insight/itkPDEDeformableRegistrationWithMaskFilter.hxx
index e50dffb..36ad568 100644
--- a/libs/demons_itk_insight/itkPDEDeformableRegistrationWithMaskFilter.hxx
+++ b/libs/demons_itk_insight/itkPDEDeformableRegistrationWithMaskFilter.hxx
@@ -19,7 +19,8 @@
 
 #include "itkPDEDeformableRegistrationWithMaskFilter.h"
 
-#include "itkExceptionObject.h"
+// #include "itkExceptionObject.h"
+#include "itkMacro.h"
 #include "itkImageRegionIterator.h"
 #include "itkImageLinearIteratorWithIndex.h"
 #include "itkDataObject.h"
diff --git a/libs/ransac/RANSAC.h b/libs/ransac/RANSAC.h
index 59fc50b..1566748 100644
--- a/libs/ransac/RANSAC.h
+++ b/libs/ransac/RANSAC.h
@@ -15,7 +15,8 @@
 #include "itkMultiThreader.h"
 #include "itkSimpleFastMutexLock.h"
 #endif
-#include "itkExceptionObject.h"
+// #include "itkExceptionObject.h"
+#include "itkMacro.h"
 
 /**
  * This class implements a multi-threaded version of the RAndom SAmple 
diff --git a/src/plastimatch/register/itk_registration_private.h b/src/plastimatch/register/itk_registration_private.h
index 3dcd585..fda1688 100644
--- a/src/plastimatch/register/itk_registration_private.h
+++ b/src/plastimatch/register/itk_registration_private.h
@@ -5,7 +5,8 @@
 #define _itk_registration_private_h_
 
 #include "plmregister_config.h"
-#include "itkExceptionObject.h"
+// #include "itkExceptionObject.h"
+#include "itkMacro.h"
 #include "itkImageRegistrationMethod.h"
 #include "itk_image_type.h"
 #include "stage_parms.h"
